osu.Game.Rulesets.Osu/Difficulty/Skills/Speed.cs

```diff
@@ -26,12 +27,13 @@ public class Speed : OsuStrainSkill
         protected override double DifficultyMultiplier => 1.04;
 
         private const double min_speed_bonus = 75; // ~200BPM
-        private const double max_speed_bonus = 45; // ~330BPM
         private const double speed_balancing_factor = 40;
+        private double greatWindow;
 
-        public Speed(Mod[] mods)
+        public Speed(Mod[] mods, float od, double clockRate)
             : base(mods)
         {
+            greatWindow = (79 - (od * 6) + 0.5) / clockRate;
```

Comment:
I think this ideally would use `OsuHitWindows` and its `SetDifficulty()` method, to avoid reimplementing the hit window formula locally.

Also, why is the `0.5` portion applied *pre*-clockrate adjustments? Was that intended?

Reply:
Agreed, I'll swap that out to using OsuHitWindows. I am going to assume that I was just incorrect on the `0.5` part of the formula.